src/main/java/com/acme/conferencesystem/cfp_proposals/business/ProposalService.java

```diff
@@ -42,9 +44,57 @@ public Proposal submitProposal(Proposal proposal) {
         return mapper.entityToProposal(submittedProposalEntity);
     }
 
+    @Override
     public Optional<Proposal> getProposalById(UUID id) {
         return repository.findById(id)
                 .map(mapper::entityToProposal);
     }
 
+    @Override
+    public void validateProposalIsAccepted(UUID proposalId) {
+        Proposal proposal = getProposalById(proposalId)
+                .orElseThrow(() -> new IllegalArgumentException("Proposal %s not found".formatted(proposalId)));
+
+        if (proposal.status() != ProposalStatus.ACCEPTED) {
+            throw new IllegalArgumentException("Proposal %s is not accepted".formatted(proposalId));
+        }
+    }
+
+    @Override
+    public void validateProposalIsNew(UUID proposalId) {
+        Proposal proposal = getProposalById(proposalId)
+                .orElseThrow(() -> new IllegalArgumentException("Proposal %s not found".formatted(proposalId)));
+
+        if (proposal.status() != ProposalStatus.NEW) {
+            throw new IllegalArgumentException("Proposal %s is not accepted".formatted(proposalId));
+        }
+    }
+
+    public void approveProposal(UUID id) {
+        repository.findById(id).ifPresent(proposalEntity -> {
+            var approvedProposal = new ProposalEntity(
+                    proposalEntity.id(),
+                    proposalEntity.title(),
+                    proposalEntity.description(),
+                    proposalEntity.speakerId(),
+                    LocalDateTime.now(),
```

Comment:
This should be updatedate not creation date

Reply:
if we want to be purist we will need a new column

Reply:
but well let's merge and we can do that in another MR

also  we can do something like this


```java
        repository.findById(id).map(proposalEntity -> new ProposalEntity(
                proposalEntity.id(),
                proposalEntity.title(),
                proposalEntity.description(),
                proposalEntity.speakerId(),
                LocalDateTime.now(),
                ProposalStatus.ACCEPTED
        )).ifPresent(repository::save);
```

Reply:
and also we can do a factory method like proposalEntity.withStatus(ACCEPTED);

Reply:
yes, I was planning to optimize the code after finishing all the pending work